### DISCUSSION TITLE
Make lifetime of "remember me" login token customizable

### DISCUSSION THREAD

**Discussion by jamsouf** [laravel/framework#33957] [General]
The "remember me" token on login has a lifetime of 5 years, which is explained in the docs as "... which will keep the user authenticated indefinitely ...".

I want to change the lifetime of this cookie, but with the built-in LoginController it's not intended by the framework to let the application change the lifetime (easily) I think.

Why it's not intended to have a config option or an easy way to change the lifetime?
What's currently the best way to do so?

**Comment by jamsouf** [comment]
@driesvints Any tip what I can do to change the cookie lifetime?

**Comment by jmsfwk** [comment]
This was implemented in #39186 for Laravel 8, but is not documented.

Adding a `remember` key to a guard definition in `auth.php` will allow you to configure the recaller cookie expiry time.

```php
'guards' => [
    'web' => [
        'driver' => 'session',
        'provider' => 'users',
        'remember' => 43200,
    ],
],
```
